aiogram/dispatcher/filters/builtin.py

```diff
@@ -491,3 +491,70 @@ def __init__(self, exception):
             return True
         except:
             return False
+
+
+class IdFilter(Filter):
+
+    def __init__(self,
+                 user_id: Optional[Union[Iterable[Union[int, str]], str, int]] = None,
+                 chat_id: Optional[Union[Iterable[Union[int, str]], str, int]] = None,
+                 ):
+        """
+        :param user_id:
+        :param chat_id:
+        """
+        if user_id is None and chat_id is None:
+            raise ValueError("Both user_id and chat_id can't be None")
+
+        self.user_id = None
+        self.chat_id = None
+        if user_id:
+            if isinstance(user_id, Iterable):
+                self.user_id = list(map(int, user_id))
+            else:
+                self.user_id = [int(user_id), ]
+        if chat_id:
+            if isinstance(chat_id, Iterable):
+                self.chat_id = list(map(int, chat_id))
+            else:
+                self.chat_id = [int(chat_id), ]
+
+    @classmethod
+    def validate(cls, full_config: typing.Dict[str, typing.Any]) -> typing.Optional[typing.Dict[str, typing.Any]]:
+        result = {}
+        if 'user' in full_config:
+            result['user_id'] = full_config.pop('user')
+        elif 'user_id' in full_config:
+            result['user_id'] = full_config.pop('user_id')
```

Comment:
Why `user_id` and `user` is separated but don't have differences?

For `chat_id` and `chat` i have the same question.

Reply:
The reason why I chose the `user_id` and `chat_id` names is that they explicitly tells the programmer what should you put as the value. I included `user` and `chat` in addition because I saw in aiogram some places, there the argument name is `user` but user_id is required. Examples: [1](https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L860), [2](https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L887) and [3](https://github.com/aiogram/aiogram/blob/c6923e331c1222b275aa87ce87ffabf17b185890/aiogram/dispatcher/dispatcher.py#L964). 

From my point of view, the arguments in examples should be renamed to `chat_id` and `user_id`, but it will break the existing code

Reply:
After discussion in [aiogram_ru#80199](https://t.me/aiogram_ru/80199) is planned to rename all `chat`&`user` arguments to `chat_id`&`user_id` with deprecation old variants until 3.0.

So please remove `chat`&`user` arguments from this place.

Reply:
Done in [8b02869](https://github.com/aiogram/aiogram/pull/172/commits/8b028693b6dfd932bef4b59e7af8e3dfe55b3e89)